mailpoet/lib/Migrations/Migration_20230215_050813.php

```diff
@@ -0,0 +1,73 @@
+<?php declare(strict_types = 1);
+
+namespace MailPoet\Migrations;
+
+use MailPoet\Migrator\Migration;
+
+class Migration_20230215_050813 extends Migration {
+  public function run(): void {
+    $this->subjectsMigration();
+  }
+
+  private function subjectsMigration(): void {
+    $this->createTable('automation_run_subjects', [
+      '`automation_run_id` int(11) unsigned NOT NULL',
+      '`key` varchar(191)',
+      '`args` longtext',
+      'index (automation_run_id)',
+    ]);
```

Comment:
I just noticed a potential problem 😬 We used to have the subjects saved as an array, thus ordered. Here, we'll lose that ability. Can that be a problem anywhere? I guess unless we allow multiple subjects of the same type, we're fine. So maybe we don't have to solve it now at all. The question is, whether we should also make `UNIQUE (automation_run_id, key)` for now...

Reply:
@websupporter Thanks for the explanation, that makes a lot of sense. We can keep it "unordered" for now, at least for the purposes of "use the first subject for this and the second for that".

But maybe for the purposes of internal API consistency, I'd add `ORDER BY id` on the query side? What I mean is that calling `getAutomationRunsForAutomation` multiple times would return the same stable results. So, adding the `id` to the query:

```sql
SELECT * FROM $subjectTable WHERE automation_run_id in (%s) order by automation_run_id, id -- added "id"
```

We won't run into multiple subjects of the same type for a long time, I guess, but it would be nice for the current ones to get consistent order in the same automation, e.g., 1. `subscriber`, 2. `segment` on every call of that method.

Reply:
Thanks @JanJakes! Added suggestion in 5d476714c